### DISCUSSION TITLE
M5Stack Core2 Buttons

### DISCUSSION THREAD

**Discussion by ritterr** [HASwitchPlate/openHASP#368] [Hardware]
Recently I bought two M5Stack Core2 devices, flashed them with openHASP and integrated them into HA. Please excuse my ignorance but does anyone know how to make the buttons (3 red circles) work in openHASP, I tried to extend the touch area by 40px in the configuration file but the touch display would not react. Given the size of the display I would really appreciate to be able to use the 3 buttons for page navigation. 

Any help would be much appreciated. 

Regards Raphael

**Comment by fvanroie** [comment]
This is tricky because LVGL doesn't have knowledge of off-screen *'buttons'*, so this would need to be handled by the touch driver.
We currently use a generic FT6336 library for this, which also doesn't take the extra area into account...
Since the buttons are not UI elements nor GPIO buttons, there is currently no way to access them in openHASP.

What do you want these buttons to do?

**Reply by ritterr** [reply]
I was planning to use them for the page navigation instead of the navigation buttons designed in LVGL. Too bad this does not work, I was planning to use the Core2 devices instead of light switches as they perfectly fit the wall box (Switzerland) but the display is really small so the additional pixels would have been great to use instead of page navigation buttons.

**Reply by nagyrobi** [reply]
Would be nice to simply be able to extend the dimensions of lvgl benyond the physical size of the screen. You'd use invisible buttons then, as usual...

**Comment by roth** [comment]
I managed to connect a Button via GPIO Settings -> Pushbutton :-) 

Is there an option to switch a page by a GPIO Pushbutton? I probably could send a mqtt message back to switch the page, but would be nicer to do this directly on the plate.. ?

**Reply by fvanroie** [reply]
Logic is left to the automation system.

**Reply by ritterr** [reply]
@roth, can you eventually share which GPIO pin(s) you used for the button(s)? Would be much appreciated!

**Reply by roth** [reply]
@ritterr I realized I just used the interrupt button and fell into the trap of believing it works.. but it does not. Sorry :(
